src/file.toit

```diff
@@ -251,9 +258,58 @@ Deletes a file, given its name.
 delete name/string -> none:
   #primitive.file.unlink
 
-/*
+/**
 Renames a file or directory.
 Only works if the $to name is on the same filesystem.
 */
 rename from/string to/string -> none:
   #primitive.file.rename
+
+/** Flag for $link indicating a hard link */
+LINK_TYPE_HARD                        ::= 0
+/** Flag for $link indicating a symbolic link on Posix */
+LINK_TYPE_SYMBOLIC                    ::= 1
+/** Flag for $link indicating a symbolic link for a file on Windows */
+LINK_TYPE_SYMBOLIC_WINDOWS_FILE       ::= 1
+/** Flag for $link indicating a symbolic link for a directory on Windows */
+LINK_TYPE_SYMBOLIC_WINDOWS_DIRECTORY  ::= 2
+
+/**
+Creates a link from $source to $target. Set type to one of the
+  LINK_TYPE_* constants. On Posix LINK_TYPE_FILE and LINK_TYPE_DIRECTORY are
+  equivalent
+*/
+link source/string target/string type/int -> none:
```

Comment:
source and target should be named arguments.

I think instead of the constants it would be nicer to have named flags.  Three functions, one with --hard, one with --soft-file, and one with --soft-directory options.  This makes the completion in the editor helpful, and the end result is readable.

Alternatively it would be nice to have one with no flags that checks whether the destination is a file, but this is a bit harder. You have to interpret the destination relative to the source in order to find what it refers to.